container/container.go

```diff
@@ -227,3 +234,51 @@ func (a *AvroContainerWriter) AddAvroContainerWriter(p *generator.Package) {
 	p.AddFunction(a.filename(), a.name(), "WriteRecord", a.writeRecordDef())
 	p.AddFunction(a.filename(), a.name(), "Flush", a.flushDef())
 }
+
+func (a *AvroContainerWriter) AddCheckSchema(p *generator.Package) {
```

Comment:
@bonds0097 Mind if this is a package function instead?

Also, instead of checking the schema every time we create a writer, can we just check it once at project start time? (With the goal being of making writer usage as simple as possible)